File: common/src/main/java/dev/mayaqq/estrogen/registry/EstrogenCreativeTab.java
package dev.mayaqq.estrogen.registry;

import dev.mayaqq.estrogen.Estrogen;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import net.minecraft.world.item.alchemy.Potion;
import net.minecraft.world.item.alchemy.PotionUtils;
import uwu.serenity.critter.api.entry.RegistryEntry;
import uwu.serenity.critter.stdlib.creativeTabs.CreativeTabRegistrar;

import static dev.mayaqq.estrogen.registry.EstrogenBlocks.*;
import static dev.mayaqq.estrogen.registry.EstrogenEntities.MOTH;
import static dev.mayaqq.estrogen.registry.EstrogenFluids.*;
import static dev.mayaqq.estrogen.registry.EstrogenItems.*;

public class EstrogenCreativeTab {

    public static final CreativeTabRegistrar TAB = CreativeTabRegistrar.create(Estrogen.REGISTRIES);

    public static final RegistryEntry<CreativeModeTab> MAIN = TAB.entry("estrogen")
        .icon(ESTROGEN_PILL::asStack)
        .displayItems(EstrogenCreativeTab::creativeTabItems)
        .register();

    public static void creativeTabItems(CreativeModeTab.ItemDisplayParameters params, CreativeModeTab.Output output) {
        output.accept(ESTROGEN_PILL);
        output.accept(CRYSTAL_ESTROGEN_PILL);
        output.accept(GENDER_CHANGE_POTION);
        output.accept(BALLS);
        output.accept(MOTH_FUZZ);
        output.accept(TESTOSTERONE_CHUNK);
        output.accept(TESTOSTERONE_POWDER);
        output.accept(USED_FILTER);
        output.accept(ESTROGEN_CHIP_COOKIE);
        output.accept(HORSE_URINE_BOTTLE);
        output.accept(ESTROGEN_PATCHES.get().getFullStack());
        output.accept(ESTROGEN_PATCHES);
        output.accept(THIGH_HIGHS);
        output.accept(MOTH_ELYTRA);
        output.accept(UWU);
        output.accept(CENTRIFUGE);
        output.accept(COOKIE_JAR);
        output.accept(DREAM_BLOCK);
        output.accept(DORMANT_DREAM_BLOCK);
        output.accept(ESTROGEN_PILL_BLOCK);
        output.accept(MOTH_WOOL);
        output.accept(QUILTED_MOTH_WOOL);
        output.accept(MOTH_WOOL_CARPET);
        output.accept(QUILTED_MOTH_WOOL_CARPET);
        output.accept(MOTH_SEAT);
        output.accept(MOTH_BED);
        output.accept(QUILTED_MOTH_BED);
        output.accept(tippedArrow(EstrogenPotions.ESTROGEN_POTION.get()));
        output.accept(MOLTEN_SLIME.getBucket());
        output.accept(TESTOSTERONE_MIXTURE.getBucket());
        output.accept(LIQUID_ESTROGEN.getBucket());
        output.accept(FILTRATED_HORSE_URINE.getBucket());
        output.accept(HORSE_URINE.getBucket());
        output.accept(MOLTEN_AMETHYST.getBucket());
        output.accept(MOTH.getSpawnEgg());
        THIGH_HIGHS.get().streamStyleItems().forEach(output::accept);
    }

    public static ItemStack tippedArrow(Potion potion) {
        ItemStack stack = new ItemStack(Items.TIPPED_ARROW);
        PotionUtils.setPotion(stack, potion);
        return stack;
    }
}

File: src/common/main/kotlin/dev/mayaqq/estrogen/content/EstrogenCreativeTab.kt
package dev.mayaqq.estrogen.content

import dev.mayaqq.estrogen.Estrogen
import dev.mayaqq.estrogen.content.EstrogenBlocks.CookieJar
import dev.mayaqq.estrogen.content.EstrogenBlocks.DreamBlock
import dev.mayaqq.estrogen.content.EstrogenBlocks.DreamCatcher
import dev.mayaqq.estrogen.content.EstrogenBlocks.EstrogenPillBlock
import dev.mayaqq.estrogen.content.EstrogenBlocks.MothBed
import dev.mayaqq.estrogen.content.EstrogenBlocks.MothCarpet
import dev.mayaqq.estrogen.content.EstrogenBlocks.MothWool
import dev.mayaqq.estrogen.content.EstrogenBlocks.QuiltedMothBed
import dev.mayaqq.estrogen.content.EstrogenBlocks.QuiltedMothCarpet
import dev.mayaqq.estrogen.content.EstrogenBlocks.QuiltedMothWool
import dev.mayaqq.estrogen.content.EstrogenFluids.FiltratedHorseUrine
import dev.mayaqq.estrogen.content.EstrogenFluids.GenderFluid
import dev.mayaqq.estrogen.content.EstrogenFluids.HorseUrine
import dev.mayaqq.estrogen.content.EstrogenFluids.LiquidEstrogen
import dev.mayaqq.estrogen.content.EstrogenFluids.MoltenAmethyst
import dev.mayaqq.estrogen.content.EstrogenFluids.MoltenSlime
import dev.mayaqq.estrogen.content.EstrogenFluids.TestosteroneMixture
import dev.mayaqq.estrogen.content.EstrogenItems.Balls
import dev.mayaqq.estrogen.content.EstrogenItems.ColonThree
import dev.mayaqq.estrogen.content.EstrogenItems.CrystalEstrogenPill
import dev.mayaqq.estrogen.content.EstrogenItems.DreamBottle
import dev.mayaqq.estrogen.content.EstrogenItems.EstrogenChipCookie
import dev.mayaqq.estrogen.content.EstrogenItems.EstrogenPatches
import dev.mayaqq.estrogen.content.EstrogenItems.EstrogenPill
import dev.mayaqq.estrogen.content.EstrogenItems.GenderChangePotion
import dev.mayaqq.estrogen.content.EstrogenItems.HorseUrineBottle
import dev.mayaqq.estrogen.content.EstrogenItems.MothElytra
import dev.mayaqq.estrogen.content.EstrogenItems.MothFuzz
import dev.mayaqq.estrogen.content.EstrogenItems.TestosteroneChunk
import dev.mayaqq.estrogen.content.EstrogenItems.TestosteronePowder
import dev.mayaqq.estrogen.content.EstrogenItems.ThighHighs
import dev.mayaqq.estrogen.id
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.core.registries.Registries
import net.minecraft.network.chat.Component
import net.minecraft.world.item.CreativeModeTab
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import net.minecraft.world.item.alchemy.Potion
import net.minecraft.world.item.alchemy.PotionUtils
import uwu.serenity.kritter.api.Registrar
import uwu.serenity.kritter.api.entry.RegistryEntry
import uwu.serenity.kritter.stdlib.creativeTab


object EstrogenCreativeTab : Registrar<CreativeModeTab> by Estrogen..Registries.CREATIVE_MODE_TAB {

    // Make it an entry here
    val Estrogen: RegistryEntry<CreativeModeTab> = creativeTab("estrogen") {
        title = Component.translatable("itemGroup.estrogen.estrogen")
        icon { EstrogenPill.defaultInstance }
        displayItems { 
            accept(EstrogenPill)
            accept(CrystalEstrogenPill)
            accept(GenderChangePotion)
            accept(Balls)
            accept(MothFuzz)
            accept(TestosteroneChunk)
            accept(TestosteronePowder)
            accept(EstrogenChipCookie)
            accept(HorseUrineBottle)
            accept(EstrogenPatches.getFullStack())
            accept(EstrogenPatches)
            accept(ThighHighs)
            accept(MothElytra)
            accept(ColonThree)
            accept(CookieJar)
            accept(DreamCatcher)
            accept(DreamBlock.asItem())
            accept(DreamBottle)
            accept(EstrogenPillBlock)
            accept(MothWool)
            accept(QuiltedMothWool)
            accept(MothCarpet)
            accept(QuiltedMothCarpet)
            accept(MothBed)
            accept(QuiltedMothBed)
            accept(tippedArrow(EstrogenPotions.EstrogenPotion))
            accept(MoltenSlime.bucket)
            accept(TestosteroneMixture.bucket)
            accept(LiquidEstrogen.bucket)
            accept(FiltratedHorseUrine.bucket)
            accept(HorseUrine.bucket)
            accept(MoltenAmethyst.bucket)
            accept(GenderFluid.bucket)
            accept(BuiltInRegistries.ITEM.get(id("moth_spawn_egg")))
            ThighHighs.styleItems.forEach(::accept)
        }
    }
}

private fun tippedArrow(potion: Potion): ItemStack = Items.TIPPED_ARROW.defaultInstance.also { PotionUtils.setPotion(it, potion) }
